pkgs/top-level/static.nix

```diff
@@ -31,7 +31,19 @@ self: super: let
     });
   };
 
-  staticAdapters = [ makeStaticLibraries propagateBuildInputs ]
+
+  # Force a "dev" output for static builds to avoid unwanted build-time noise in "out".
+  # Add a "dev" output for static builds in case there are `propagatedBuildInputs`.
+  # Do not add "dev" if it already exists.
+  addDevOutput = stdenv: stdenv // {
+    mkDerivation = args: stdenv.mkDerivation (args // {
+      outputs = if stdenv?outputs then
```

Comment:
Also, maybe *only* do this if there are propagated dependencies? We don't want this on e.g. fixed output derivations or `runCommand`!

Reply:
Right, that bit got lost when I rewrote it